File: docs/breez-sdk/snippets/wasm/connect.ts
import {
  Breez,
  type BreezSdk,
  type SdkCredentials,
  type ConnectOptions
} from '@breeztech/breez-sdk-spark'

// Init stub
const init = async () => {}

// ANCHOR: connect-simple
const connectSimple = async (): Promise<BreezSdk> => {
  await init()

  const credentials: SdkCredentials = {
    type: 'mnemonic',
    apiKey: '<breez api key>',
    mnemonic: '<mnemonic words>',
    passphrase: undefined
  }
  const sdk = await Breez.connect(credentials)
  return sdk
}
// ANCHOR_END: connect-simple

// ANCHOR: connect-with-options
const connectWithOptions = async (): Promise<BreezSdk> => {
  await init()

  const credentials: SdkCredentials = {
    type: 'mnemonic',
    apiKey: '<breez api key>',
    mnemonic: '<mnemonic words>',
    passphrase: undefined
  }
  const options: ConnectOptions = {
    network: 'regtest',
    storageDir: './.data'
  }
  const sdk = await Breez.connect(credentials, options)
  return sdk
}
// ANCHOR_END: connect-with-options

export { connectSimple, connectWithOptions }

File: docs/breez-sdk/snippets/rust/src/connect.rs
use anyhow::Result;
use breez_sdk_spark::*;

// ANCHOR: connect-simple
pub(crate) async fn connect_simple() -> Result<BreezSdk> {
    let sdk = Breez::connect(
        SdkCredentials::Mnemonic {
            api_key: "<breez api key>".to_string(),
            mnemonic: "<mnemonic words>".to_string(),
            passphrase: None,
        },
        None,
    )
    .await?;
    Ok(sdk)
}
// ANCHOR_END: connect-simple

// ANCHOR: connect-with-options
pub(crate) async fn connect_with_options() -> Result<BreezSdk> {
    let sdk = Breez::connect(
        SdkCredentials::Mnemonic {
            api_key: "<breez api key>".to_string(),
            mnemonic: "<mnemonic words>".to_string(),
            passphrase: None,
        },
        Some(ConnectOptions {
            network: Some(Network::Regtest),
            storage_dir: Some("./.data".to_string()),
            ..Default::default()
        }),
    )
    .await?;
    Ok(sdk)
}
// ANCHOR_END: connect-with-options

// ANCHOR: connect-with-providers
pub(crate) async fn connect_with_providers() -> Result<BreezSdk> {
    let providers = Providers {
        // storage: Some(my_custom_storage),
        // chain_service: Some(my_custom_chain_service),
        ..Default::default()
    };

    let sdk = Breez::with_providers(providers)
        .connect(
            SdkCredentials::Mnemonic {
                api_key: "<breez api key>".to_string(),
                mnemonic: "<mnemonic words>".to_string(),
                passphrase: None,
            },
            None,
        )
        .await?;
    Ok(sdk)
}
// ANCHOR_END: connect-with-providers
